src/main/drivers/pwm_output_dshot_hal.c

```diff
@@ -82,6 +89,12 @@ FAST_CODE void pwmCompleteDshotMotorUpdate(uint8_t motorCount)
 {
     UNUSED(motorCount);
 
+    /* If there is a dshot command loaded up, time it correctly with motor update*/
+    if (pwmProcessDshotCommand(motorCount)) {
```

Comment:
Almost there. :wink: 

This will still call `pwmProcessDshotCommand()` in slow RAM for every motor update. Instead the call should be

```
if (pwmIsProcessingDshotCommand()) {
    if (pwmProcessDshotCommand(motorCount)) {
        return;
    }
}
```

this way, only (non time critical) Dshot commands will run from slow RAM.

Also, I'd probably swap the return values in `pwmProcessDshotCommand()`, so that the above becomes

```
if (pwmIsProcessingDshotCommand()) {
    if (!pwmProcessDshotCommand(motorCount)) {
        return;
    }
}
```

which looks a bit more intuitive.

Reply:
This still needs to be fixed, or else it will not fit into ITCM_RAM.